GANDLF/cli/config_generator.py

```diff
@@ -1,11 +1,15 @@
 import yaml
+from typing import Union
 from pathlib import Path
 from copy import deepcopy
 
 
 def generate_new_configs_from_key_and_value(
-    base_config, key, value, upper_level_key=None
-):
+    base_config: dict,
+    key: str,
+    value: Union[str, list, int],
+    upper_level_key: str = None,
```

Comment:
In the cases when the param is typed to string and default set to None, as in the previous cases it should be Union[str,None]

Reply:
Fixed with `Optional`.